examples/block_send.rs

```diff
@@ -101,20 +101,37 @@ fn create_meta(id: ProtocolId) -> (ProtocolMeta, Arc<AtomicUsize>) {
 
 fn main() {
     env_logger::init();
+    let mut rt = tokio::runtime::Runtime::new().unwrap();
 
     if std::env::args().nth(1) == Some("server".to_string()) {
-        let (meta, _) = create_meta(1.into());
-        let mut service = create(false, meta, ());
-        let listen_addr = service
-            .listen("/ip4/127.0.0.1/tcp/8900".parse().unwrap())
-            .unwrap();
-        println!("listen_addr: {}", listen_addr);
-        tokio::run(service.for_each(|_| Ok(())));
+        rt.block_on(async move {
+            let (meta, _) = create_meta(1.into());
+            let mut service = create(false, meta, ());
+            let listen_addr = service
+                .listen("/ip4/127.0.0.1/tcp/8900".parse().unwrap())
+                .await
+                .unwrap();
+            println!("listen_addr: {}", listen_addr);
+            loop {
+                if service.next().await.is_none() {
+                    break;
+                }
+            }
+        });
     } else {
-        let listen_addr = std::env::args().nth(1).unwrap().parse().unwrap();
-        let (meta, _result) = create_meta(1.into());
-        let mut service = create(false, meta, ());
-        service.dial(listen_addr, DialProtocol::All).unwrap();
-        tokio::run(service.for_each(|_| Ok(())));
+        rt.block_on(async move {
+            let listen_addr = std::env::args().nth(1).unwrap().parse().unwrap();
+            let (meta, _result) = create_meta(1.into());
+            let mut service = create(false, meta, ());
+            service
+                .dial(listen_addr, TargetProtocol::All)
+                .await
+                .unwrap();
+            loop {
+                if service.next().await.is_none() {
+                    break;
+                }
+            }
```

Comment:
Should be `service.for_each().await;` , otherwise it will not concurrent.

Many other places as well.